apps/framework-cli/src/framework/typescript/parser.rs

```diff
@@ -92,17 +92,37 @@ fn extract_data_models_from_ast(ast: Module) -> Result<FileObjects, TypescriptPa
     Ok(FileObjects::new(parsed_models, enums))
 }
 
-fn enum_to_data_enum(enum_decl: &TsEnumDecl) -> DataEnum {
+fn enum_to_data_enum(enum_decl: TsEnumDecl) -> Result<DataEnum, TypescriptParsingError> {
     let name = enum_decl.id.sym.to_string();
-    let values = enum_decl
-        .members
-        .iter()
-        .map(|member| match &member.id {
+    let mut values = Vec::new();
+
+    for member in enum_decl.members {
+        let name = match &member.id {
             TsEnumMemberId::Ident(ident) => ident.sym.to_string(),
             TsEnumMemberId::Str(str) => str.value.to_string(),
-        })
-        .collect();
-    DataEnum { name, values }
+        };
+
+        let value = match member.init {
+            Some(init) => match *init {
+                Expr::Lit(Lit::Str(str)) => EnumValue::String(str.value.to_string()),
+                Expr::Lit(Lit::Num(num)) => EnumValue::Int(num.value as u8),
+                _ => {
+                    return Err(TypescriptParsingError::UnsupportedDataTypeError {
+                        type_name: format!("{:?}", init),
+                    })
+                }
+            },
+            None => {
+                return Err(TypescriptParsingError::UnsupportedDataTypeError {
+                    type_name: "Non mapped enums".to_string(),
+                })
+            }
+        };
+
+        values.push(EnumMember { name, value });
```

Comment:
If the enum values are strings, I don't think the `name` is meaningful anywhere outside TS.

Reply:
Clickhouse uses it to map the values. Ref: https://clickhouse.com/docs/en/sql-reference/data-types/enum

Reply:
```
enum Direction {
  Up = "UP",
  Down = "DOWN",
  Left = "LEFT",
  Right = "RIGHT",
}
```

`Right` is just only in js/ts. In JSON it's `RIGHT`, so in clickhouse it should be the string values as well.

Reply:
Yeah but in the case for numbers 

```
enum Direction {
  Up = 0,
  Down = 1,
  Left = 2,
  Right = 3,
}
```

it is better to have clickhouse with `Enum('Up' = 0, 'Down' = 1...)` that way the user has human readable labels